### ISSUE TITLE
Picture loaded and loaderror events should not be attempted if the chart has been disposed 

### ISSUE THREAD

**Issue by DavidAtImpactCubed** [amcharts/amcharts5#1355]
We have a chart on our site that, in some conditions, gets disposed quite quickly after being rendered. This chart contains several Pictures. Sometimes, the chart is disposed before the pictures have finished loading.

In version 5.8.3, a fix was introduce: "Picture's events "loaded" and "loaderror" were kicking in before actual picture was fully loaded/errored."
https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md#583---2024-02-19

An unintended consequence of this fix for us is that we now get exceptions thrown on the check to see if the event is enabled:
```
				if (this.events.isEnabled("loaded")) {
					this.events.dispatch("loaded", { type: "loaded", target: this });
				}
```

The proposed fix is to check if the event dispatcher is disposed before attempting to check if the event isEnabled:
```
				if (!this.events.isDisposed() && this.events.isEnabled("loaded")) {
					this.events.dispatch("loaded", { type: "loaded", target: this });
				}
```

Thank you for considering this fix!

https://github.com/amcharts/amcharts5/blame/d4e20ac481f37bbfec3370606a5326137e9c9a91/src/.internal/core/render/Picture.ts#L156

**Comment by DavidAtImpactCubed** [comment]
Issue was accidentally closed as it was referenced in one of our private repositories! Apologies.

**Comment by martynasma** [comment]
Fixed in [5.8.5](https://github.com/amcharts/amcharts5/releases/tag/5.8.5).

## [5.8.5] - 2024-03-15

### Added
- `forceHidden` (default: `false`) setting added to `StockControl`. Enables forcing some `StockChart` controls to be always hidden.
- New `DateRangeSelector` event added: `"rangeselected"`.
- New `PriodSelector` event added: `"periodselected"`.

### Fixed
- A JavaScript error was being triggered if a `Picture` was disposed before external image was finished loading (since 5.8.3).
- Inputs in `DateRangeSelector` were ignoring `Root`'s `timezone` and `utc` settings.
- Thicker stroke (`strokeWidth > 1`) was being ignored as a hit target for interactive elements.
- `Scrollbar` grips would not update their position according to `maxZoomCount`/`minZoomCount` setting in some cases.
- Hovering over the border of a `Treemap` node could result tooltip of some other node to appear.
- In some cases pointer over tooltip of a hovered `Treemap` node would remain on the same node, even if pointer would move to some other node.
- If a `{ timeUnit: "week", count: 2 }` (count > 1) was added to a `DateAxis`, labels were flickering while scrolling.
- Calling `zoomToDates()`/`zoomToValues()` on a `DateAxis` was resulting not exact dates when data grouping was enabled and grouping period would changed during zoom.

[Full change log](https://github.com/amcharts/amcharts5/blob/master/packages/shared/CHANGELOG.md).

[Download options](https://www.amcharts.com/download/).

Make sure you clear your browser cache after upgrading. And feel free to contact us again if you are still experiencing this issue.
